__tests__/fdir.test.ts

```diff
@@ -370,6 +370,30 @@ for (const type of apiTypes) {
       .crawl("node_modules");
     t.expect(globFunction).toHaveBeenCalledWith(["**/*.js"], "bleep");
   });
+
+  test(`[${type}] using custom fs implementation`, async (t) => {
+    const readdirStub = vi.fn<Parameters<typeof fs.readdir>>((_path, _opts, cb) => {
+      cb(null, []);
+    });
+    const readdirSyncStub = vi.fn();
+    readdirSyncStub.mockReturnValue([]);
+    const fakeFs = {
+      ...fs,
+      readdir: readdirStub,
+      readdirSync: readdirSyncStub
+    } as unknown as typeof fs;
+
+    const api = new fdir({
+        fs: fakeFs
+      })
+      .crawl("node_modules");
+    await api[type]();
+    if (type === 'withPromise') {
```

Comment:
this might need to be double quotes to align with the project's style. i'm in the habit of using single quotes too though :smile:

Reply:
Surprised it doesn't have a formatter setup, unless I just missed it somehow

If that doesn't exist, I'll open a separate pr to add prettier